### ISSUE TITLE
ListObjects returns invalid key 

### ISSUE THREAD

**Issue by philipmarivoet** [minio/minio#9121]
## Expected Behavior

ListObjects should never return invalid keys

## Current Behavior

Bucket containing one object (key: A/B)

ListObjects using a marker A/C pointing after the last key in the keyspace returns a non empty result set containing a key which is pointing to a directory A/

## Steps to Reproduce (for bugs)

Following test reproduces the problem.
```
public class MinioTest
{
    @Test
    public void testListObjectsReturnsInvalidKey()
    {
        String bucketName = "bucket";
        AmazonS3 s3;

        AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
        ...
        s3 = builder.build();
        s3.createBucket(bucketName);
        s3.putObject(bucketName, "A/B", "content");

        ListObjectsRequest request = new ListObjectsRequest().withBucketName(bucketName)
                                                             .withMaxKeys(10)
                                                             .withMarker("A/C");

        ObjectListing result = s3.listObjects(request);

        assertNotNull(result);
        assertFalse(result.isTruncated());
        assertTrue(result.getObjectSummaries().isEmpty());
    }
}
```

Result contains one entry:

```
S3ObjectSummary{
    bucketName='bucket', 
    key='A/', 
    eTag='d41d8cd98f00b204e9800998ecf8427e', 
    size=0,
    lastModified=Wed Mar 11 16:24:51 CET 2020, 
    storageClass='STANDARD', 
    owner=S3Owner[
        name=,id=02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4
    ]
}

```

## Your Environment
* Version used: RELEASE.2020-03-09T18-26-53Z

**Comment by harshavardhana** [comment]
This is as per S3 behavior, your marker is lexically higher than your key. So you are not interested in listing A/B ?

**Comment by philipmarivoet** [comment]
Indeed, I expect no keys to be returned as there do not exist any keys after the specified marker.
However the response does contains a key "A/" which does not exist in the bucket.

Other S3 implementations we tested (AWS, ECS and IBM) do return the correct response (no keys present).

**Comment by harshavardhana** [comment]
> Other S3 implementations we tested (AWS, ECS and IBM) do return the correct response (no keys present).

Can you provide the sample response from AWS? @philipmarivoet

**Comment by harshavardhana** [comment]
We were able to reproduce this, thank you for opening the issue. We also have a fix already, should be available in the next release.

**Comment by harshavardhana** [comment]
Fixed by #9129
